serval-agent/src/api/mod.rs

```diff
@@ -7,31 +7,35 @@ use axum::{
 use http::header::HeaderValue;
 use serde::{Deserialize, Serialize};
 use tokio::sync::Mutex;
-use utils::blobs::BlobStore;
+use utils::{blobs::BlobStore, errors::ServalError};
 use uuid::Uuid;
 
-use std::collections::HashMap;
 use std::sync::Arc;
+use std::{collections::HashMap, path::PathBuf};
 
 pub mod jobs;
 pub mod storage;
 
 #[derive(Debug, Clone, Serialize)]
 pub struct RunnerState {
-    storage: BlobStore,
+    storage: Option<BlobStore>,
     jobs: HashMap<String, JobMetadata>,
     total: usize,
     errors: usize,
 }
 
 impl RunnerState {
-    pub fn new(blob_path: String) -> Self {
-        RunnerState {
-            storage: BlobStore::new(blob_path.into()),
+    pub fn new(blob_path: Option<PathBuf>) -> Result<Self, ServalError> {
+        let storage = match blob_path {
```

Comment:
If there's a more idiomatic way to do this, I'd love to hear it. I tried variants along these lines...

````rust
let storage = blob_path.map(|path| BlobStore::new(path)?);
````

...but `Option` and `Result` don't mix super well.